### PR TITLE
feat: Add fragment links for easy linking of page subsections

### PR DESCRIPTION
**Pull request by rschristian**
Makes for an easier time of linking to subsections as users (likely us, most of the time) won't have to grab the link by first using the table of contents.

Limited to the guide at the moment; could support the tutorial, but would need a bit different styling as we can't get away with the negative margin there.

### CODE REVIEW

File: src/lib/content.js [hunk 1]
```diff
@@ -118,6 +119,14 @@ function parseMarkdownContent(data) {
 	});
 }
 
+function addFragmentLinksToHeadings(data) {
+	data.html = data.html.replaceAll(
+		/(<h[1-6] id="([^"]+)">)([^\n]+)/g,
+		'$1<a href="#$2"></a>$3'
+	);
+	return data;
+}
```
Comment (marvinhagemeister) [inline src/lib/content.js:128]: If we upgrade `marked` we can hook directly in their markdown renderer instead of regexing the HTML after all is done.

Comment (rschristian) [reply src/lib/content.js:128]: Oh neat, didn't realize it supported that. Will take a look tomorrow

Comment (rschristian) [reply src/lib/content.js:128]: Oh man, 9 major versions. Will need time to verify everything's working correctly

Comment (rschristian) [reply src/lib/content.js]: So `marked` has tore through 5 majors in the past 4 months, including ripping some things we use. I don't think it makes sense to touch that now, especially as the docs don't seem to be in a good state.

**Comment by developit** [comment]
This is a good idea. I agree with avoiding the Marked upgrade, that sounds way too painful.

As an experiment, we could use the HTML-element-to-component mapping layer we have in place to remap `h[1-6]` to components that render the heading and link as a pair:
https://github.com/preactjs/preact-www/pull/1013

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**